lib/machine/machine.ts

```diff
@@ -15,26 +15,134 @@
+    configureGoals(sdm);
+    configureGenerators(sdm);
+    configureExtensionPacks(sdm);
+    configureBuilder(sdm);
+
+    if (isInLocalMode()) {
+        configureDeploysForLocalSdm(sdm);
+    } else {
+        configureDeploysForCloudSdm(sdm);
+    }
 
     return sdm;
 }
+
+export function configureDeploysForCloudSdm(sdm: SoftwareDeliveryMachine) {
+    configureLocalSpringBootDeploy(sdm);
+    sdm.addDeployRules(
+        deploy.when(IsMaven)
+            .deployTo(StagingDeploymentGoal, StagingEndpointGoal, StagingUndeploymentGoal)
+            .using(
+                {
+                    deployer: localExecutableJarDeployer(),
+                    targeter: ManagedDeploymentTargeter,
+                },
+            ),
+    );
+
+    sdm.addDisposalRules(
+        whenPushSatisfies(AnyPush)
+            .itMeans("We can always delete the repo")
+            .setGoals(RepositoryDeletionGoals));
+
+    sdm.addCommand(EnableDeploy)
+        .addCommand(DisableDeploy)
+        .addCommand(DisplayDeployEnablement);
+}
+
+function configureGoals(sdm: SoftwareDeliveryMachine) {
+    // Each contributor contributes goals. The infrastructure assembles them into a goal set.
+    sdm.addGoalContributions(goalContributors(
+        onAnyPush().setGoals(new Goals("Checks", ReviewGoal, PushReactionGoal, AutofixGoal)),
+        whenPushSatisfies(IsDeploymentFrozen)
+            .setGoals(ExplainDeploymentFreezeGoal),
+        whenPushSatisfies(anySatisfied(IsMaven))
+            .setGoals(JustBuildGoal),
+    ));
+}
+
+function configureGenerators(sdm: SoftwareDeliveryMachine) {
+    sdm
+        .addGeneratorCommand<SpringProjectCreationParameters>({
+            name: "create-spring",
+            intent: "create spring",
+            description: "Create a new Java Spring Boot REST service",
+            paramsMaker: SpringProjectCreationParameters,
+            startingPoint: new GitHubRepoRef("spring-team", "spring-rest-seed"),
+            transform: [
+                ReplaceReadmeTitle,
+                SetAtomistTeamInApplicationYml,
+                TransformSeedToCustomProject,
+            ],
+        })
+        .addGeneratorCommand<SpringProjectCreationParameters>({
+            name: "create-spring-kotlin",
+            intent: "create spring kotlin",
+            description: "Create a new Kotlin Spring Boot REST service",
+            paramsMaker: SpringProjectCreationParameters,
+            startingPoint: new GitHubRepoRef("johnsonr", "flux-flix-service"),
+            transform: [
+                ReplaceReadmeTitle,
+                SetAtomistTeamInApplicationYml,
+                TransformSeedToCustomProject,
+            ],
+        });
+}
+
+function configureExtensionPacks(sdm: SoftwareDeliveryMachine) {
+    sdm.addExtensionPacks(
+        SpringSupport,
+    );
+}
+
+export function configureBuilder(sdm: SoftwareDeliveryMachine) {
+    const mb = new MavenBuilder(sdm);
+    sdm.addBuildRules(
```

Comment:
`addBuildRules` is deprecated too. Please register goal implementations directly.

Reply:
Replaced with goal implementations